docs/development_environment.md

```diff
@@ -0,0 +1,230 @@
+
+![Reopen in container](assets/dev-in-vscode/vsc-reopen-in-con.png)
+
+Notice the little green box now specifies that the editor is open in a docker container:
+
+![VSCode in container](assets/dev-in-vscode/vsc-in-container.png)
+
+## 5. Setup Python Development Environment
+
+The `setup.py` does all the hard work of installing linters and testing software...
+
+### 5.1 Install the Python Extension
+
+Navigate to the extensions panel on the left bar once again and search for the Python extension. You'll notice that it must be installed again, but this time, it must be installed on the remote VSCode server. Click to install it. After installing, reload the editor (the install button will have turned into a blue reload button).
+
+![Install Python](assets/dev-in-vscode/vsc-install-python.png)
+
+### 5.2 Settings
+
+In the root project directory, create a `.vscode` directory, in that crate a `settings.json` file with the following contents:
+```json
+{
+    "terminal.integrated.shell.linux": "/bin/zsh",
+    "python.pythonPath": "/venv/bin/python3",
+    "python.linting.pylintEnabled": true,
+    "python.linting.pylintPath": "/venv/bin/pylint",
+    "python.linting.enabled": true,
+    "python.testing.pytestArgs": [],
+    "python.testing.unittestEnabled": false,
+    "python.testing.nosetestsEnabled": false,
+    "python.testing.pytestEnabled": true
+}
+```
+
+This will enable the python extension, debugging, linting and unit testing facilities.
+
+### 5.3 Run Config
+
+Going one step further, we can setup some default run configurations so that when `F5` is pressed, the server is debuggable from within the editor. To crate a run config to start the server, in the `.vscode` directory, create a new file called `launch.json` with the following contents:
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Python: Flask",
+            "type": "python",
+            "request": "launch",
+            "module": "flask",
+            "env": {
+                "FLASK_APP": "elpis",
+                "FLASK_ENV": "development",
+                "FLASK_DEBUG": "0"
+            },
+            "args": [
+                "run",
+                "--host","0.0.0.0",
+                "--port","5000",
+                "--no-debugger",
+                "--no-reload"
+            ],
+            "jinja": true
+        }
+    ]
+}
+```
+
+Now press `F5` (or the play button) and see the server run. Try using breakpoints to pause the server at a line of code. Try the `elpis/elpis/__init__.py` file pause the program around line 72 on the print statement in the index function. To get the server to pause you will need to open a browser and load `0.0.0.0:5000` to run that section of code.
```

Comment:
<img width="698" alt="Screen Shot 2020-09-22 at 10 34 46 am" src="https://user-images.githubusercontent.com/9084563/93834465-5dbf6980-fcbf-11ea-87ef-e04baf0a742a.png">
It doesn't appear to stop at this breakpoint.

Reply:
Ah yep, that documetnation is wrong, I'll update it, it should say browse to `0.0.0.0:5000/index.html`